bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/TypeChecker.java

```diff
@@ -2799,15 +2806,17 @@ private static boolean checkFiniteTypeAssignable(Object sourceValue, Type source
     protected static boolean isFiniteTypeValue(Object sourceValue, Type sourceType, Object valueSpaceItem,
                                                boolean allowNumericConversion) {
         Type valueSpaceItemType = getType(valueSpaceItem);
-        if (valueSpaceItemType.getTag() > TypeTags.DECIMAL_TAG) {
-            return valueSpaceItemType.getTag() == sourceType.getTag() &&
+        sourceType = TypeUtils.getReferredType(sourceType);
```

Comment:
since `sourceType` is not directly used anywhere in this function shall we make a variable for `int sourceTypeTag = TypeUtils.getReferredType(sourceType).getTag();

Reply:
Updated thanx